### DISCUSSION TITLE
What happened to github.event.head_commit.modified?

### DISCUSSION THREAD

**Discussion by pheel** [community/community#25597] [Actions]
<p>Out of the box, GitHub Actions makes it possible to run workflows conditionally on the modification of a path:</p>
<pre><code>on: push: paths: -'foo/\*\*/\*'
</code></pre>
<p>I have the vague memory that there was a time where <code>github.event.head_commit.modified</code> resolved to an array of modified paths. Which would allow:</p>
<pre><code>jobs:
  job1:
    if: contains(github.event.head_commit.modified, 'foo/package.json')
    ...
steps:
- if: contains(github.event.head_commit.modified, 'foo/bar/package.json')
</code></pre>
<p>I created <a href="https://github.com/marketplace/actions/path-watcher-action" rel="nofollow noopener">a simple action</a> that queries the <a href="https://developer.github.com/v3/repos/commits/#get-a-single-commit" rel="nofollow noopener"><code>/repos/:owner/:repo/commits/:ref</code></a> endpoint and determines whether a path was modified.</p>
<p>It’s easy enough to use that action to impose conditions on running a <em>step</em>, but I would like to put conditions on <em>jobs</em> inside a workflow. Since there is a <em>steps</em>, but not a <em>jobs</em> context, this is not straightforward. Any idea of how I could achieve this result?</p>

**Comment by peter-evans** [comment, marked as answer]
<p>That attribute was removed. See the <a href="https://github.blog/changelog/2019-10-16-changes-in-github-actions-push-event-payload/" rel="nofollow noopener">blog post here</a>.</p>
<p>I don’t know how to add to properties in the job context of the current workflow, but what you could do is trigger a repository_dispatch event and send whatever context you need to a second workflow. This could include the output of your action. Then it would be available immediately and useable via job conditions in the second workflow. <a href="https://github.com/peter-evans/repository-dispatch" rel="nofollow noopener">This action</a> might help if you want to try that idea.</p>

**Comment by pheel** [comment]
<p>Very nice, thanks. I’ll let others chip in but this seems an appropriate workaround.</p>
